scripts/o3de/o3de/cmake.py

```diff
@@ -27,75 +29,55 @@ def add_gem_dependency(cmake_file: pathlib.Path,
     """
     adds a gem dependency to a cmake file
     :param cmake_file: path to the cmake file
-    :param gem_name: name of the gem
+    :param gem_name: name of the gem with optional version specifier
     :return: 0 for success or non 0 failure code
     """
     if not cmake_file.is_file():
         logger.error(f'Failed to locate cmake file {str(cmake_file)}')
         return 1
 
-    # on a line by basis, see if there already is {gem_name}
-    # find the first occurrence of a gem, copy its formatting and replace
-    # the gem name with the new one and append it
-    t_data = []
-    added = False
-    start_marker_line_index = None
-    end_marker_line_index = None
+    gem_name_with_version_specifier = gem_name
+    gem_name, _ = utils.get_object_name_and_optional_version_specifier(gem_name)
+
+    pre_gem_names = ''
+    post_gem_names = ''
+    all_gem_names = '' 
+    indent = 4
+
     with cmake_file.open('r') as s:
-        in_gem_list = False
-        line_index = 0
-        for line in s:
-            parsed_line = line.strip()
-            if parsed_line.startswith(enable_gem_start_marker):
-                # Skip pass the 'set(ENABLED_GEMS' marker just in case their are gems declared on the same line
-                parsed_line = parsed_line[len(enable_gem_start_marker):]
-                # Set the flag to indicate that we are in the ENABLED_GEMS variable
-                in_gem_list = True
-                start_marker_line_index = line_index
+        file_contents = s.read()
+        if file_contents:
+            # regex to isolate pre, gem names and post 
+            regex_all_gem_names = re.compile('(?P<pre>[\s\S]*set\s*\(\s*ENABLED_GEMS\s*)?(?P<gem_names>[^\)]+)(?P<post>\)[\s\S]*)?')
```

Comment:
This sequence being matched in the `pre` and `post` named capture group checks doesn't make any sense.
This is matching zero or more characters that **is whitespace** character or **is not a whitespace**.
In that case `.` can be used.

Reply:
The dot character matches newlines as well as long it doesn't used the `$` anchor at the end: https://regex101.com/r/pGU4Bn/1

Reply:
It does not seem to capture multiple lines, i.e. everything up until `ENABLED_GEMS` https://regex101.com/r/nDfF0a/1

Reply:
It looks like the dot for python matches zero or more characters that are not line terminators
> matches any character (except for line terminators)
![image](https://user-images.githubusercontent.com/56135373/223563293-b8a19a29-37af-4b05-b8a9-8dd5df853570.png)

So I guess the regular expression you have is a funky way to match multiple lines, though I don't think that was the intent of the regular expression engine.

Reply:
well, I'm currently looking into remove the need for `enable_gems.cmake` if that goes well I'll revert these changes

Reply:
I've removed this function and related tests as it is no longer necessary to add gems to `enabled_gems.cmake`